### PR TITLE
[Optimze] Adding and removing use flags and deps

### PR DESCRIPTION
**Pull request by croessner**
* Removing gmime
	* Added pcre2 support
	* Added jemalloc support
	* Add PAX hardening routine for mprotect flag

### CODE REVIEW

**Comment by djc** [comment]
I'd like to keep the latest "normal" ebuild mostly in sync with the -9999 one; I think most of your improvements here also apply to the 1.4 branch, right? Can you please make sure they're applied to both? Also in the other direction; I set +fann in 1.4.1-r1 to make it more similar to upstream, so that should be in 9999, too (not doing so is my mistake, but since you're working on this area anyway...)

**Comment by croessner** [comment]
I just chked out git and switched to tag 1.4.1. I can find JEMALLOC and PCRE2 also in this CMakeLists.txt, so I am pretty sure, it should also apply to 1.4.x as well :-)

gmime was dropped in 1.5.0pre (current master).

File: mail-filter/rspamd/rspamd-9999.ebuild [hunk 1]
```diff
@@ -13,23 +13,35 @@ EGIT_REPO_URI="https://github.com/vstakhov/rspamd.git"
...
-IUSE="fann +gd +jit libressl"
+IUSE="fann +gd pcre2 +jit jemalloc libressl"
 
 RDEPEND="!libressl? ( dev-libs/openssl:0[-bindist] )
 		libressl? ( dev-libs/libressl:0 )
 		fann? ( sci-mathematics/fann )
 		jit? (
-			dev-libs/libpcre[jit]
+			pcre2? (
+				dev-libs/libpcre2[jit]
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-9999.ebuild]: shouldn't it suffice to drop the jit? clauses and just do
```
	pcre2? (
		dev-libs/libpcre2[jit=]
	)
	!pcre2? (
		dev-libs/libpcre[jit=]
	)
```

Comment (croessner) [inline mail-filter/rspamd/rspamd-9999.ebuild]: Thank you :-) I fixed it in this branch.

Comment (croessner) [inline mail-filter/rspamd/rspamd-9999.ebuild]: "This branch" means my repo->PR-branch

**Comment by djc** [comment]
So maybe I wasn't clear: *in this pull request*, please also revbump the 1.4.1 ebuild and make it match the 9999 ebuild as closely as possible, then I can merge them together.

**Comment by croessner** [comment]
@djc Shouldn't I create a -r2 then?

**Comment by croessner** [comment]
@djc I just see that you already added PAX rules. But you do PAX not dependend on "jit", why?

File: mail-filter/rspamd/rspamd-9999.ebuild [hunk 2]
```diff
@@ -53,6 +54,8 @@ src_configure() {
 		-DENABLE_LUAJIT=$(usex jit ON OFF)
 		-DENABLE_FANN=$(usex fann ON OFF)
 		-DENABLE_GD=$(usex gd ON OFF)
+		-DENABLE_PCRE2=$(usex pcre2 ON OFF)
+		-DENABLE_JEMALLOC=$(usex jemalloc ON OFF)
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-9999.ebuild]: in theory you can drop all these ON OFF thingies, as cmake interprets `yes` and `no` too (https://cmake.org/cmake/help/v3.0/command/if.html)

File: mail-filter/rspamd/rspamd-9999.ebuild [hunk 3]
```diff
@@ -61,6 +64,11 @@ src_install() {
 	cmake-utils_src_install
 	newinitd "${FILESDIR}/rspamd.init-r2" rspamd
 
+	# Remove mprotect for JIT support
+	if use jit; then
+		pax-mark m "${ED}"/usr/bin/rspamd-* "${ED}"/usr/bin/rspamadm-*
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-9999.ebuild]: missing `|| die`

**Comment by djc** [comment]
Yes, you should bump it to -r2 (that's why I mentioned "revbump"). I likely forgot to make the PAX marking dependent on the `jit` flag, please fix that too. :)

**Comment by croessner** [comment]
I just pushed a 1.4.1-r2. Can you review it?

Concerning the ON/OFF, I currently would stay with this notation, as you never know, what future Cmake-changinges might bring. At least, it seems more cleaner to me :-)

**Comment by SoapGentoo** [comment]
@croessner while I can't vouch for Kitware, it's extremely unlikely that they will change this, as it's part of the API now and changing this would cause widespread breakage, but your choice.

File: mail-filter/rspamd/rspamd-9999.ebuild (hunk 3 above)
Comment (croessner) [inline mail-filter/rspamd/rspamd-9999.ebuild]: Can you help me pointing to the right location, where you miss the || die exactly?

Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-9999.ebuild]: `pax-mark m "${ED%/}"/usr/bin/rspamd-* "${ED%/}"/usr/bin/rspamadm-* || die`

**Comment by croessner** [comment]
I added the "die" statements

@SoapGentoo How can I fix the commit messages now?

**Comment by SoapGentoo** [comment]
@croessner `git rebase` is your biggest friend: https://wiki.gentoo.org/wiki/Gentoo_Github#Rebasing_commits
You can also use rebase to **reword** specific COMMITs. Also squash some of your commits

**Comment by croessner** [comment]
@djc I want to be careful here. I see that I need to use squash instead of fixup, but the docs do not tell me, if I have to modify the resulting commit message in this editor window, or if it will be asked after saving this.

I have this right now:

pick c21ed7f5d8 [Optimze] Adding and removing use flags and deps
s 0566821164 [Fix] Simplified USE flag dependencies
s 6877f09753 [Merge] Adding new features from live ebuild
s c9a3c59881 [Fix] Adding die statements

Do I change the commit message here or later?

**Comment by SoapGentoo** [comment]
```
r c21ed7f [Optimze] Adding and removing use flags and deps
s 0566821 [Fix] Simplified USE flag dependencies
s 6877f09 [Merge] Adding new features from live ebuild
s c9a3c59 [Fix] Adding die statements
```
Will merge everything into one commit, for which you can edit the **COMMITMSG**

**Comment by croessner** [comment]
I have rebased it and I hope it follows your guidelines.

**Comment by SoapGentoo** [comment]
@croessner you forgot the colon after CAT/PN :smile: 
rebase again or use `git commit --amend`

**Comment by croessner** [comment]
Two little dots. ha ha :-) Fixed

File: mail-filter/rspamd/rspamd-1.4.1-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+SRC_URI="https://rspamd.com/downloads/${P}.tar.xz"
+HOMEPAGE="https://github.com/vstakhov/rspamd"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="fann pcre2 +jit jemalloc libressl"
+
+RDEPEND="!libressl? ( dev-libs/openssl:0[-bindist] )
+		libressl? ( dev-libs/libressl:0 )
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-1.4.1-r2.ebuild]: openssl and libressl need `:0=`

File: mail-filter/rspamd/rspamd-1.4.1-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+		jemalloc? (
+			dev-libs/jemalloc
+		)
+		dev-libs/libevent
+		dev-db/sqlite:3
+		dev-libs/glib:2
+		dev-libs/gmime
+		dev-util/ragel
+		sys-apps/file
+		virtual/libiconv"
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-1.4.1-r2.ebuild]: too much indentation (the baseline should be one tab)

File: mail-filter/rspamd/rspamd-1.4.1-r2.ebuild [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+inherit cmake-utils pax-utils user systemd
+
+DESCRIPTION="Rapid spam filtering system"
+SRC_URI="https://rspamd.com/downloads/${P}.tar.xz"
+HOMEPAGE="https://github.com/vstakhov/rspamd"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="fann pcre2 +jit jemalloc libressl"
```
Comment (SoapGentoo) [inline mail-filter/rspamd/rspamd-1.4.1-r2.ebuild]: sort these alphabetically please

**Comment by croessner** [comment]
I hope I have fixed both PRs correctly. @SoapGentoo thank you very much for your support on making this ebuild better. And @djc, too, of course ;-)

**Comment by djc** [comment]
Unless I'm mistaken, this PR now contains only the 9999 changes, not the revised 1.4.2 ebuild. Can you fix that up, please? I'm otherwise ready to merge this!

**Comment by djc** [comment]
I asked you to put that in this PR, did you miss that, or just disagree? :)

**Comment by croessner** [comment]
Hm 😊 I removed 1.4.1-r2 from the first PR, so I thought this was ok. Do I really need to merge and rebase again?

**Comment by djc** [comment]
Rebased all your changes as 46c0780bef4f21a9683afce5cfba8ac7f66fdde1 and fixed up the formatting in 6f9483bb00abf01a13bec7cf2d9c5ff69b9b7ee9.

**Pull request closed without merging**